### PR TITLE
Upgrading to jdk-23+26.

### PR DESCRIPTION
**Pull request by dbalek**
Upgrading to jdk-23+26

### CODE REVIEW

**Comment by jtulach** [comment]
There is a [failure on JDK 17](https://github.com/JaroslavTulach/nb-javac/actions/runs/9479101517/job/26120478560#step:10:18).

File: make/langtools/netbeans/nb-javac/nbproject/project.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
-jdk.git.url=https://github.com/openjdk/jdk22u
-jdk.git.commit=jdk-22.0.1+8
-nb-javac-ver=${jdk.git.commit}
-#nb-javac-ver=jdk-21u
+jdk.git.url=https://github.com/openjdk/jdk
+jdk.git.commit=jdk-23+26
+#nb-javac-ver=${jdk.git.commit}
+nb-javac-ver=dev
```
Comment (jtulach) [inline make/langtools/netbeans/nb-javac/nbproject/project.properties]: Dušane, why do you set the nb-javac-ver to dev? Does not it [produce strange results](https://github.com/JaroslavTulach/nb-javac/actions/runs/9480584618/job/26121592268?pr=23#step:7:224) then? Do we really want to publish `nb-javac-dev-impl.jar` on Maven central?

Comment (dbalek) [reply make/langtools/netbeans/nb-javac/nbproject/project.properties]: You are right. Fixed.

Comment (jtulach) [reply make/langtools/netbeans/nb-javac/nbproject/project.properties]: Looks [good](https://github.com/JaroslavTulach/nb-javac/actions/runs/9495124378/job/26166753760?pr=23#step:7:223).

**Comment by jtulach** [comment]
Staged at https://oss.sonatype.org/content/repositories/comdukescript-1166

**Pull request merged**